src/main/java/com/logpresso/scanner/Log4j2Scanner.java

```diff
@@ -611,19 +653,28 @@ private Status scanNestedJar(File fatJarFile, ZipFile zipFile, ZipEntry zipEntry
 				if (entry == null)
 					break;
 
-				if (entry.getName().equals(LOG4j_CORE_POM_PROPS))
+				if (entry.getName().equals(LOG4j_CORE_POM_PROPS)) {
 					vulnerableVersion = loadVulnerableLog4jVersion(zis);
+				}
 
-				if (entry.getName().equals(JNDI_LOOKUP_CLASS_PATH))
+				if (entry.getName().equals(JNDI_LOOKUP_CLASS_PATH)) {
 					mitigated = false;
+					foundJndiClass = entry.getName().equals(LOG4j_CORE_POM_PROPS);
+				}
 			}
 
 			if (vulnerableVersion != null) {
 				String path = fatJarFile + " (" + zipEntry.getName() + ")";
-				printDetection(path, vulnerableVersion, mitigated);
+				printDetection(path, vulnerableVersion, mitigated, false);
 				return mitigated ? Status.MITIGATED : Status.VULNERABLE;
 			}
 
+			if(foundJndiClass) {
```

Comment:
Use `!mitigated` instead of `foundJndiClass`. This version doesn't detect nested potentially vulnerable jar.

Reply:
Hhmm.
Nested potentially vulnerable jars are found in my tests. I need to know if the neesed JAR contains the pom and is safe; otherwise 2.15er versions (vulnerableVersion == null) will also be marked as potentially vulnerable.